v3/src/models/formula/utils/name-mapping-utils.ts

```diff
@@ -69,17 +69,19 @@ export const getDisplayNameMap = (options: IDisplayNameMapOptions, useSafeSymbol
   displayNameMap.localNames.caseIndex = localAttrIdToCanonical(CASE_INDEX_FAKE_ATTR_ID)
 
   dataSets.forEach(dataSet => {
-    // No LOCAL_ATTR prefix is necessary for external attributes. They always need to be resolved manually by custom
-    // mathjs functions (like "lookupByIndex"). Also, it's never necessary to use safe names, as these names
-    // are string constants, not symbols, so MathJS will not care about special characters there.
-    const dataSetKey = key(dataSet.name, false)
-    displayNameMap.dataSet[dataSetKey] = {
-      id: idToCanonical(dataSet.id),
-      attribute: {}
+    if (dataSet.title) {
+      // No LOCAL_ATTR prefix is necessary for external attributes. They always need to be resolved manually by custom
+      // mathjs functions (like "lookupByIndex"). Also, it's never necessary to use safe names, as these names
+      // are string constants, not symbols, so MathJS will not care about special characters there.
+      const dataSetKey = key(dataSet.title, false)
+      displayNameMap.dataSet[dataSetKey] = {
+        id: idToCanonical(dataSet.id),
+        attribute: {}
+      }
```

Comment:
This seems fine to fix the bug, but since we're here I think it would make sense to include the `name` in the `displayNameMap` as well, something like:
```suggestion
      displayNameMap.dataSet[key(dataSet.title, false)] = {
        id: idToCanonical(dataSet.id),
        attribute: {}
      }
      if (dataSet.name !== dataSet.title) {
        displayNameMap.dataSet[key(dataSet.name, false)] = {
          id: idToCanonical(dataSet.id),
          attribute: {}
        }
      }
```
So we're following the general principle that we should match on either wherever possible.